app/database/queries/breeze_buddy/credentials.py

```
@@ -0,0 +1,143 @@
+    """
+    Generate query to get credentials for a merchant.
+    If merchant_id is provided, returns merchant-specific + global credentials.
+    If merchant_id is None, returns only global credentials.
+    """
+    if merchant_id:
+        text = f"""
+            SELECT * FROM "{CREDENTIALS_TABLE}"
+            WHERE ("merchant_id" = $1 OR "merchant_id" IS NULL)
+            AND "is_active" = TRUE
+            ORDER BY "merchant_id" NULLS FIRST, "name" ASC;
+        """
```

Ruff (0.15.1)

[error] Possible SQL injection vector through string-based query construction

(S608)